src/app/app.ts

```diff
@@ -26,17 +24,33 @@ class AppComponent {
   }
 
   public newSubscription() {
-    this.channels.push(this.newSearchTerm);
+    this.channels.push({term: this.newSearchTerm, active: true});
     this.newSearchTerm = '';
   }
 
   public clearSearch(channel) {
-    this.channels = this.channels.filter(function(ch) {
-      return ch !== channel;
+    this.channels = this.channels.filter((ch) => {
+      if (ch.term === channel.term) {
+        // unsubscribe when clearing
+        this.toggleSearch(channel);
+      }
+      return ch.term !== channel.term;
     });
   }
-  public stopSearch(channel) {
-    this.pusher.unsubscribe(btoa(channel));
+  public toggleSearch(channel) {
+    for (let ch of this.channels) {
+      if (ch.term === channel.term) {
+        if (ch.active) {
+          ch.active = false;
+          this.pusher.unsubscribe(btoa(channel.term));
+        } else {
+          ch.active = true;
+          var encoded = btoa(channel.term);
```

Comment:
If anything maybe should be: `var` => `let`.

I'll push something even better in moment though.